tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +58,44 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experiment$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    })
+  );
+
+  private readonly experimentName$ = this.experiment$.pipe(
+    map((experiment) => (experiment ? experiment.name : null))
+  );
+
+  readonly title$ = this.store.select(getEnvironment).pipe(
+    combineLatestWith(this.experimentName$),
+    map(([env, experimentName]) => {
+      if (env.window_title && env.window_title.length > 0) {
+        // (it's an empty string when the `--window_title` flag is not set)
+        return env.window_title;
+      } else if (experimentName && experimentName.length > 0) {
+        return experimentName;
```

Comment:
I'd consider always using the brand name (as discussed below) as a suffix to the title, so that the full title clearly indicates this is TensorBoard no matter what strange user-provided experiment name there might be.  This is pretty common in various other places where the primary title is user-controlled content, e.g. github pull request pages are suffixed with the repo name, google docs pages end in ` - Google Docs`, google search results end in ` - Google Search`.

So maybe `${experimentName} - ${this.tbBrandName}` here?

Reply:
Thanks for the suggestion!